### PR TITLE
Remove redundant code paths in call_op_set.h

### PR DESCRIPTION
**Pull request by vjpai**
Compared to other PRs in this project, this has a truly minor impact on code size, dropping about 3K (< .5%)

### CODE REVIEW

File: include/grpcpp/impl/codegen/call_op_set.h [hunk 1]
```diff
@@ -499,7 +506,7 @@ class CallOpRecvMessage {
   ByteBuffer recv_buf_;
   bool allow_not_getting_message_ = false;
   bool hijacked_ = false;
-  bool hijacked_recv_message_failed_ = false;
```
Comment (yashykt) [inline include/grpcpp/impl/codegen/call_op_set.h:502]: why was this changed?

Comment (vjpai) [reply include/grpcpp/impl/codegen/call_op_set.h:502]: I can remove this from the PR if you think it's important. My feeling was to save this instruction from the constructor and instead only bring it in for situations where the variable is actually used. I don't think this change had any impact on code size so I'm ok with reverting it from this PR.

Comment (vjpai) [reply include/grpcpp/impl/codegen/call_op_set.h]: reverted these lines

Comment (yashykt) [reply include/grpcpp/impl/codegen/call_op_set.h]: I thought the style guide mandated initialization but if it doesn't, I dont mind the change.

**Review by yashykt — APPROVED** [review]
LGTM. Please squash and merge.

File: include/grpcpp/impl/codegen/call_op_set.h (hunk 1 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/call_op_set.h]: I also confirmed that this change had no impact on code size, so it doesn't fit with this PR anyway. We can revisit it if it's valuable for performance in another PR.

**Comment by vjpai** [comment]
Thanks for the reviews!
